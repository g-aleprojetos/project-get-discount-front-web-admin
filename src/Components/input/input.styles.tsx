import styled from 'styled-components';
import cores from 'resources/colors';

export type PropsInput = React.InputHTMLAttributes<HTMLInputElement>;

export const ContainerInput = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 20px;
`;
export const LabelBox = styled.label`
  margin-left: 10px;
  font-size: 12px;
  text-transform: uppercase;
  color: ${cores.shuttleGray};
`;

export const InputBox = styled.input`
  height: 30px;
  margin-top: 5px;
  border-radius: 8px;
  border: 1px solid transparent;
  font-size: 16px;
  background-color: ${cores.shuttleGray};
  color: ${cores.white};
  outline: 0;
`;
