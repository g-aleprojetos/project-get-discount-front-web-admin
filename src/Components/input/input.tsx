import React from 'react';
import * as S from './input.styles';

type Props = S.PropsInput & {
  testId?: string;
  label?: string;
  value?: string;
  requered?: boolean;
  handleOnchange: (event: string) => void;
};

export const Input = (props: Props) => {
  const {testId, label, id, value, requered = false, handleOnchange} = props;

  return (
    <S.ContainerInput data-testid={testId ?? 'input-pardao'}>
      <S.LabelBox data-testid="input-label" htmlFor={id}>
        {label}
      </S.LabelBox>
      <S.InputBox
        data-testid="input-padrao-box"
        id={id}
        value={value}
        required={requered === true ? true : false}
        onChange={event => {
          handleOnchange(event.currentTarget?.value);
        }}
        {...props}
      />
    </S.ContainerInput>
  );
};
