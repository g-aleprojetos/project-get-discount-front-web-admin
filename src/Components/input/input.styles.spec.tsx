import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from './input.styles';

describe('Input.styles', () => {
  test('ContainerInput DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerInput />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('LabelBox DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.LabelBox />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('InputBox DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.InputBox />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
