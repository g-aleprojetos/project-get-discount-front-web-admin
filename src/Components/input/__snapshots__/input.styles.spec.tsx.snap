// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.styles ContainerInput DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 20px;
}

<div
  className="c0"
/>
`;

exports[`Input.styles InputBox DEVE ser igual ao snapshot 1`] = `
.c0 {
  height: 30px;
  margin-top: 5px;
  border-radius: 8px;
  border: 1px solid transparent;
  font-size: 16px;
  background-color: #545d6a;
  color: #ffffff;
  outline: 0;
}

<input
  className="c0"
/>
`;

exports[`Input.styles LabelBox DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin-left: 10px;
  font-size: 12px;
  text-transform: uppercase;
  color: #545d6a;
}

<label
  className="c0"
/>
`;
